src/sequence_batch_scheduler.cc

```diff
@@ -1573,7 +1777,7 @@ DirectSequenceBatch::BatcherThread(const int nice)
           std::unique_lock<std::mutex> lk(payload_mu_);
           exec_complete_ = true;
         }
-        payload_cv_.notify_one();
+        payload_cv_.notify_all();
```

Comment:
Why the move from `notify_one` to `notify_all` based on a single payload's readiness?

Reply:
The condition variable has two different points where it is waited on, the batcher and destructor both wait until the last enqueued payload is complete. When the batcher thread is being destructed, notify just one might only unblock the batcher thread, but not the destructor, so notify all will ensure everything unblocks.